hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRequestFutureImpl.java

```diff
@@ -989,7 +989,7 @@ private void invokeCallBack(byte[] regionName, byte[] row, CResult result) {
   }
 
   private void cleanServerCache(ServerName server, Throwable regionException) {
-    if (ClientExceptionsUtil.isMetaClearingException(regionException)) {
+    if (tableName == null && ClientExceptionsUtil.isMetaClearingException(regionException)) {
```

Comment:
In which case tableName could be null?

Reply:
It's only null for HTableMultiplexer

Reply:
This reverts the logic back to pre-21775: https://github.com/apache/hbase/commit/5f25985b98f8ed36a6c5cc48c16b0621c60d8769#diff-7c58ffd83c150488599591ed5a3a068599646ebdbbbfdcd2233386e5472cca35L921

The `tableName == null` check here was standing for very long prior to that. I don't think it was correct to remove, since in the `tableName != null` (most usages) we'd fall into updateCachedLocations which clears the cache just for the individual regions.

Reply:
@Apache9 can you give another look?

Reply:
@saintstack Do you still remember something sir? Skimmed the patch, I'm not sure why removing the `tableName == null` check can fix the problem described in HBASE-21775, as Tommy Li said

> from what I can see, tableName shouldn't be null unless you manually create a BufferedMutatorImpl instead of using ConnectionFactory.createConnection().getBufferedMutator(). I not sure if the bufferedmutator would work at all without a table name.